ros/src/common/cmake/autoware_build_flags/cmake/autoware_build_flags-extras.cmake

```diff
@@ -9,4 +9,24 @@ if(${CMAKE_VERSION} VERSION_LESS "3.1.0")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 else()
   set(CMAKE_CXX_STANDARD 11)
-endif()
\ No newline at end of file
+endif()
+
+find_package(CUDA QUIET)
```

Comment:
Only search for CUDA if the user has explicitly enabled it. i.e. something like:

    USE_CUDA=FALSE
    if(ENABLE_CUDA)
      if ("$ENV{ROS_DISTRO}" STREQUAL "melodic")
        find_package(CUDA REQUIRED)
        if (${CUDA_VERSION} etc. etc.)
          USE_CUDA=TRUE
        endif()
      else()
        message(FATAL "CUDA support requires ROS Melodic or greater")
      endif()
    else()
      message(STATUS "CUDA support is disabled. Use -DENABLE_CUDA=TRUE and install Eigen >= 3.3.7 to enable it")
    endif()

Then you can also just do a check on `USE_CUDA` rather than needing to check both `ENABLE_CUDA` and `CUDA_FOUND` elsewhere.

Reply:
got it, preparing a new PR